### ISSUE TITLE
Unable to add downloads to orders in admin dashboard

### ISSUE THREAD

**Issue by jrick1229** [woocommerce/woocommerce#29679]
**Prerequisites (mark completed items with an [x]):**
- [x] I have have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

**Describe the bug**

When trying to manually add a downloadable product to an order (download, not the product line item) in the 'Downloadable product permissions' box, clicking 'Grant access' results in an infinite spin.

There are no console errors logged.

If trying to add downloads to a pre-existing order, this fatal error is logged after refreshing or updating the page:

```
2021-04-14T17:23:37+00:00 CRITICAL Uncaught Error: Call to a member function get_quantity() on null in /Users/joey/Local Sites/testsite10/app/public/wp-content/plugins/woocommerce/includes/class-wc-ajax.php:808
Stack trace:
#0 /Users/joey/Local Sites/testsite10/app/public/wp-includes/class-wp-hook.php(292): WC_AJAX::grant_access_to_download('')
#1 /Users/joey/Local Sites/testsite10/app/public/wp-includes/class-wp-hook.php(316): WP_Hook->apply_filters('', Array)
#2 /Users/joey/Local Sites/testsite10/app/public/wp-includes/plugin.php(484): WP_Hook->do_action(Array)
#3 /Users/joey/Local Sites/testsite10/app/public/wp-admin/admin-ajax.php(187): do_action('wp_ajax_woocomm...')
#4 {main}
  thrown in /Users/joey/Local Sites/testsite10/app/public/wp-content/plugins/woocommerce/includes/class-wc-ajax.php on line 808
```

**Expected behavior**

Prior to v5.2.0 I don't remember seeing this behavior, though I would expect that adding downloads via that box would be possible.

**Actual behavior**

Unable to add downloads to a product in the dashboard, even if you add the product to the order while in Pending Payment as well.
I've tested with different order statuses, but nothing seems to change the inability to add those downloads.

**Steps to reproduce the bug (We need to be able to reproduce the bug in order to fix it.)**

Step to reproduce the bug:
1. Create an order from either within the dashboard or normal checkout
2. Make sure you've fully saved the order
3. Enter the product name into the 'Downloadable product permissions' input
4. Click 'Grant access' - you'll see the infinite spin
5. Now either 'Update' the order or 'Refresh' the page
6. Go to: WooCommerce > Status > Logs > [fatal-errors-...]
7. See the error I mentioned above

**Screenshots**

Infinite spin: https://d.pr/i/QCb3dg
Consistent errors in testing: https://d.pr/i/pYEm0N
Download settings: https://d.pr/i/IRQTpm (changing these made no difference)
Product settings: https://d.pr/i/PJFFRy

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**WordPress Environment**

<details>

```
### WordPress Environment ###

WordPress address (URL): http://testsite10.local
Site address (URL): http://testsite10.local
WC Version: 5.2.0
REST API Version: ✔ 5.2.0
WC Blocks Version: ✔ 4.7.2
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 2.1.5
Log Directory Writable: ✔
WP Version: 5.7
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.16.0
PHP Version: 7.3.5
PHP Post Max Size: 1,000 MB
PHP Time Limit: 1200
PHP Max Input Vars: 4000
cURL Version: 7.54.0
LibreSSL/2.6.5

SUHOSIN Installed: –
MySQL Version: 8.0.16
Max Upload Size: 300 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 5.2.0
WC Database Prefix: wp_
Total Database Size: 90.87MB
Database Data Size: 70.79MB
Database Index Size: 20.08MB

### Active Plugins (3) ###

WooCommerce Stripe Gateway: by WooCommerce – 5.1.0
WooCommerce: by Automattic – 5.2.1
WP Mail Logging: by MailPoet – 1.9.7

### Theme ###

Name: Storefront
Version: 3.5.0 (update to version 3.5.1 is available)
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –
```
</details>

**Comment by jrick1229** [comment]
3908691-zen

**Comment by tammullen** [comment]
Hi @jrick1229 ,

Thank you for taking the time to report this bug, we really appreciate your help. I can reproduce it on my end. 


![Screenshot 2021-04-14 at 21 37 44](https://user-images.githubusercontent.com/24649833/114777663-cc533a00-9d6b-11eb-8d8a-9a7db0945156.png)
I confirmed this works in 5.1.0

![Screenshot_2021-04-14_at_21_47_51](https://user-images.githubusercontent.com/24649833/114777594-c2c9d200-9d6b-11eb-9be7-3a248876785e.png)
![Screenshot_2021-04-14_at_21_50_49](https://user-images.githubusercontent.com/24649833/114777613-c52c2c00-9d6b-11eb-9165-95c4f58e09ae.png)
In 5.2.1 UI freezes and there is a 500 error in console and critical error in log

We’re going to solve this bug as soon as possible. Stay tuned for updates.

**Comment by tammullen** [comment]
I think this might be due to the a [fix](https://github.com/woocommerce/woocommerce/pull/29631) included in 5.2.1 for [this PR](https://github.com/woocommerce/woocommerce/pull/23188) which is not in 5.2.1 and is milestoned for 5.3.0

**Comment by solstudioim** [comment]
Affected user with similar user 3882438-zen

```
The site works perfectly now_, but *the "Grant download" no longer
works, the wheel rotates forever.*
```

**Comment by Konamiman** [comment]
Fixed in https://github.com/woocommerce/woocommerce/pull/29691 and released in WooCommerce 5.2.2
